lib/remap-lang/src/value.rs

```diff
@@ -999,21 +999,25 @@ impl Value {
 
         let mut handle_field = |field: &Field, new| {
             let key = field.as_str().to_owned();
-            let mut map = BTreeMap::default();
+
+            // `handle_field` is used to update map values, if the current value
+            // isn't a map, we need to make it one.
+            if !matches!(self, Value::Map(_)) {
+                *self = BTreeMap::default().into()
+            }
+
+            let map = match self {
```

Comment:
Clearly it is working, but I don't quite understand why this doesn't have to be `let mut`?

Reply:
`self` is already a `&mut self` in this case, so that's what we're getting back from the inner value as well.